### PR TITLE
Allow using a static password for DB logins

### PR DESCRIPTION
**Pull request by hexedpackets**
Clients can now be authenticated using a username of the client ID and a user-defined password. Passwords do not have an expiration but MUST be linked to a single manifest. The manifest ID is the only piece of metadata lost compared to the encoded tokens.

### CODE REVIEW

**Comment by debussyman** [comment]
One question before reviewing, why do passwords have to be unique per manifest? Aren't clients themselves unique per manifest?

File: lib/jumpwire/client_auth.ex [hunk 1]
```diff
@@ -92,4 +112,57 @@ defmodule JumpWire.ClientAuth do
...
+
+  ## Token
+
+  The username is assumed to be the ID of the manifest for the connection.
+  The password is a token encoded and signed by the `Plug.Crypto` module.
+  The signature of the token is verified and metadata is pulled out of the
+  decoded token to find the relevant ClientAuth.
+  """
+  def verify(username, password) do
+    case fetch(username) do
```
Comment (debussyman) [inline lib/jumpwire/client_auth.ex:133]: Does it matter that this isn't validating the org that the client belongs to?

Comment (hexedpackets) [reply lib/jumpwire/client_auth.ex:133]: I don't think so. A malicious authn attempt would need to know both the ID and the password - also having the org id doesn't add much protection IMO.

**Comment by hexedpackets** [comment]
> One question before reviewing, why do passwords have to be unique per manifest? Aren't clients themselves unique per manifest?

There's no requirement for passwords to be unique. The requirement is that ClientAuths that have a password set must have a manifest_id set as well - you can't authenticate with a ClientAuth that is allowed to access every manifest in the org.

**Review by debussyman — APPROVED** [review]

**Pull request merged**